x/superfluid/keeper/msg_server_test.go

```diff
@@ -192,3 +193,191 @@ func (suite *KeeperTestSuite) TestMsgLockAndSuperfluidDelegate() {
 		}
 	}
 }
+
+// TestMsgSuperfluidDelegate_Event tests that events are correctly emitted
+// when calling SuperfluidDelegate.
+func (suite *KeeperTestSuite) TestMsgSuperfluidDelegate_Event() {
+	type param struct {
+		lockOwner sdk.AccAddress
+		duration  time.Duration
+	}
+
+	tests := []struct {
+		name       string
+		param      param
+		expectPass bool
+	}{
+		{
+			name: "basic valid",
+			param: param{
+				lockOwner: sdk.AccAddress([]byte("addr1---------------")), // setup wallet
+				duration:  time.Hour * 504,
+			},
+			expectPass: true,
+		},
+		{
+			name: "invalid duration",
+			param: param{
+				lockOwner: sdk.AccAddress([]byte("addr1---------------")), // setup wallet
+				duration:  time.Second,
+			},
+			expectPass: false,
+		},
+	}
+
+	for _, test := range tests {
+		suite.SetupTest()
+
+		denoms, _ := suite.SetupGammPoolsAndSuperfluidAssets([]sdk.Dec{sdk.NewDec(20), sdk.NewDec(20)})
+		coinsToLock := sdk.NewCoins(sdk.NewCoin(denoms[0], sdk.NewInt(20)))
+		suite.FundAcc(test.param.lockOwner, coinsToLock)
+
+		lockupMsgServer := lockupkeeper.NewMsgServerImpl(suite.App.LockupKeeper)
+		c := sdk.WrapSDKContext(suite.Ctx)
+		resp, err := lockupMsgServer.LockTokens(c, lockuptypes.NewMsgLockTokens(test.param.lockOwner, test.param.duration, coinsToLock))
+
+		valAddrs := suite.SetupValidators([]stakingtypes.BondStatus{stakingtypes.Bonded})
+
+		msgServer := keeper.NewMsgServerImpl(suite.App.SuperfluidKeeper)
+		_, err = msgServer.SuperfluidDelegate(c, types.NewMsgSuperfluidDelegate(test.param.lockOwner, resp.ID, valAddrs[0]))
+
+		if test.expectPass {
+			suite.Require().NoError(err)
+			assertEventEmitted(suite, suite.Ctx, types.TypeEvtSuperfluidDelegate, 1)
```

Comment:
why do you need to pass `Suite` and `Suite.Ctx`? You can just get the context if you pass `Suite`